### PR TITLE
[React] Extended notifications

### PR DESCRIPTION
**Pull request by imobachgs**
This PR is basically an evolution of #319. I have dropped cogo-toast (nice project, btw) and implemented a small *duck* to handle notifications and adding a new component to display them. Moreover, I have added timeout support.

Pending stuff (separate PRs)

- [x] Good looking notifications :wink:
  
  new-notifications png

- [x] Allow the user to block a given type of notifications ("do not show again")

### CODE REVIEW

File: src/components/Notifications/Notification.js [hunk 1]
```diff
@@ -21,9 +21,12 @@ function Notification({ notification }) {
       {text}
 
       <div className="mt-2">
-        <button className="text-blue-600 font-bold mr-2" onClick={() => dispatch(actions.close(id))}>
+        <button className="text-blue-600 font-bold mr-2" onClick={() => dispatch(actions.close(notification.id))}>
           Dismiss
         </button>
+        <button className="text-blue-600 font-bold mr-2" onClick={() => dispatch(actions.blacklistAndClose(notification))}>
```
Comment (dgdavid) [inline src/components/Notifications/Notification.js:27]: For now, I'd use `text-blue-300` for this action.

https://tailwindcss.com/docs/customizing-colors#default-color-palette

File: src/state/ducks/notifications.js [hunk 2]
```diff
@@ -10,23 +10,112 @@
  *
  * @todo Remember which notifications should be 'silenced'.
```
Comment (ancorgs) [inline src/state/ducks/notifications.js:11]: This is not longer `@todo` isn't it?

File: src/components/Notifications/Notification.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      className="w-1/3 mb-2 p-3 text-sm text-gray-700 border-b-2 border-secondary bg-white shadow-lg"
+    >
+      {text}
+
+      <div className="mt-2">
+        <button className="text-blue-600 font-bold mr-2" onClick={() => dispatch(actions.close(notification.id))}>
+          Dismiss
+        </button>
+
+        {notification.type &&
```
Comment (ancorgs) [inline src/components/Notifications/Notification.js:28]: I guess we can refine this in the future, not sure whether having type or not is the right criteria for allowing the "do not show again".

Comment (imobachgs) [reply src/components/Notifications/Notification.js]: Yes, you are right. I came to the same conclusion. But now that I've added support for notification actions, this code is not relevant anymore :)

**Comment by imobachgs** [comment]
I am not happy about the fact of having to pass the notification ID to the `dispatchAction` action creator. I added such an ID to the action itself, but then I changed my mind. By now, I think it is good enough. We can refine later.

File: src/state/ducks/notifications.test.js [hunk 4]
```diff
@@ -68,26 +74,14 @@ describe('action creators', () => {
     });
   });
 
-  describe('notifyMessage', () => {
-    it('adds a message and removes it after a timeout', () => {
+  describe('blacklist', () => {
+    it('adds the message type to the black list', () => {
       const store = mockStore({ notifications: initialState });
-      store.dispatch(actionCreators.notifyMessage('You have been muted.'));
+      console.log("ACTIOH", actionCreators.blacklist('muted'));
```
Comment (dgdavid) [inline src/state/ducks/notifications.test.js:80]: A debugging leftover?

Comment (imobachgs) [reply src/state/ducks/notifications.test.js]: :flushed:

**Pull request merged**